src/wp-admin/includes/privacy-tools.php

```diff
@@ -381,19 +378,35 @@ function wp_privacy_generate_personal_data_export_file( $request_id ) {
 		),
 	);
 
-	// Merge in the special about group.
-	$groups = array_merge( array( 'about' => $about_group ), $groups );
+	// And now, all the Groups.
+	$groups = get_post_meta( $request_id, '_export_data_grouped', true );
 
-	$groups_count = count( $groups );
+	// Merge in the special about group.
+	if ( is_array( $groups ) ) {
+		$groups       = array_merge( array( 'about' => $about_group ), $groups );
+		$groups_count = count( $groups );
+	} elseif ( '' === $groups || false === $groups ) {
+		$groups       = array( 'about' => $about_group );
+		$groups_count = 1;
+	} else {
+		wp_send_json_error(
+			sprintf(
+				'Warning: array_merge(): Expected parameter 2 to be an array, %1$s given.',
+				gettype( $groups )
+			)
+		);
```

Comment:
Preserves the original warning when/if `'_export_data_grouped'` exists and is not an `array` type.

- < PHP 7.3
>Warning: array_merge(): Argument #2 is not an array

- PHP 7.3-7.4
>Warning: array_merge(): Expected parameter 2 to be an array, string given

- PHP 8
>Fatal error: Uncaught TypeError: array_merge(): Argument #2 must be of type array, string given

See it in action here https://3v4l.org/VlCAl